### PR TITLE
fix(persons): un-ratelimit person destroy, use kafka producer for messages

### PR DESCRIPTION
**Pull request by macobo**
## Problem

In https://github.com/PostHog/posthog/pull/11576, we rate limited person destroy endpoint since a customer was causing us issues by consistently spamming the endpoint. 

The root cause was sync_execute-ing INSERT queries, each of which creates a separate part in clickhouse. Once we hit a parts limit, all hell breaks loose
## Changes

- Refactor code to use existing methods (which use KafkaProducer) to perform deletions
- Remove rate limits

During this discovered another bug: https://github.com/PostHog/posthog/issues/11590

### CODE REVIEW

File: posthog/api/person.py [hunk 1]
```diff
@@ -64,8 +64,7 @@
 from posthog.queries.stickiness import Stickiness
 from posthog.queries.trends.lifecycle import Lifecycle
 from posthog.queries.util import get_earliest_timestamp
-from posthog.rate_limit import DestroyClickhouseModelThrottle
-from posthog.settings import EE_AVAILABLE, RATE_LIMIT_ENABLED
+from posthog.settings import EE_AVAILABLE
```
Comment (neilkakkar) [inline posthog/api/person.py:67]: nit: Use `from django.conf import settings` to get settings. This ensures you can easily override settings in tests, is the recommended way, and means you don't have to search for the right file for the setting when using a folder-style settings config

File: posthog/models/person/person.py [hunk 2]
```diff
@@ -69,14 +69,11 @@ def split_person(self, main_distinct_id: Optional[str]):
 
                 from posthog.models.person.util import create_person, create_person_distinct_id
 
-                create_person_distinct_id(
```
Comment (neilkakkar) [inline posthog/models/person/person.py:72]: How is splitting distinctIDs handled now?

**Review by neilkakkar — APPROVED** [review]
Makes a lot of sense

File: posthog/api/person.py (hunk 1 above)
Comment (neilkakkar) [reply posthog/api/person.py:67]: oh, this is unrelated to current changes

File: posthog/models/person/person.py (hunk 2 above)
Comment (macobo) [reply posthog/models/person/person.py:72]: Mostly correctly. Note that the previous code didn't actually do anything in this branch for pdi2 table.

**Pull request merged**